lisa/sut_orchestrator/azure/transformers.py

```diff
@@ -219,16 +239,35 @@ def _export_vhd(self, platform: AzurePlatform, virtual_machine: Any) -> str:
         )
 
         if runbook.custom_blob_name:
-            path = runbook.custom_blob_name
+            vhd_path = runbook.custom_blob_name
+            vmgs_path = runbook.custom_blob_name.replace(".vhd", "_vmgs.vhd")
+        else:
+            vhd_path, vmgs_path = _generate_vhd_path(
+                container_client, runbook.file_name_part
+            )
+        vhd_url_path = f"{container_client.url}/{vhd_path}"
+        vhd_blob_client = container_client.get_blob_client(vhd_path)
+        vhd_blob_client.start_copy_from_url(
+            sas_url, metadata=None, incremental_copy=False
+        )
+
+        if vmgs_sas_url:
+            assert vmgs_path != vhd_path, (
+                "vmgs url path is the same as vhd url path. "
+                "Make sure the custom_blob_name ends in .vhd"
+            )
+            vmgs_url_path = f"{container_client.url}/{vmgs_path}"
+            vmgs_blob_client = container_client.get_blob_client(vmgs_path)
+            vmgs_blob_client.start_copy_from_url(
+                vmgs_sas_url, metadata=None, incremental_copy=False
+            )
+            wait_copy_blob(vmgs_blob_client, vmgs_url_path, self._log)
         else:
-            path = _generate_vhd_path(container_client, runbook.file_name_part)
-        vhd_path = f"{container_client.url}/{path}"
-        blob_client = container_client.get_blob_client(path)
-        blob_client.start_copy_from_url(sas_url, metadata=None, incremental_copy=False)
+            vmgs_url_path = ""
 
-        wait_copy_blob(blob_client, vhd_path, self._log)
+        wait_copy_blob(vhd_blob_client, vhd_url_path, self._log)
```

Comment:
Add comments to explain the reason to wait here, instead of earlier.

Reply:
It allows the copy of VMGS and VHD to happen in parallel. I will add a comment.